src/persist/collect-idb-databases.ts

```diff
@@ -0,0 +1,162 @@
+  void sleepFiveAndCollect(idbDatabasesStore, signal);
+
+  const stopInterval = initBgIntervalProcess(
+    'CollectIDBDatabases',
+    async () => {
+      await collectIDBDatabases(
+        idbDatabasesStore,
+        signal,
+        Date.now(),
+        MAX_AGE,
+        undefined,
+      );
+    },
+    COLLECT_INTERVAL_MS,
+    lc,
+  );
+
+  signal.addEventListener('abort', stopInterval);
+}
+
+async function sleepFiveAndCollect(
+  idbDatabasesStore: IDBDatabasesStore,
+  signal: AbortSignal,
+) {
+  try {
+    await sleep(COLLECT_DELAY, signal);
+
+    await collectIDBDatabases(
+      idbDatabasesStore,
+      signal,
+      Date.now(),
+      MAX_AGE,
+      defaultNewDagStore,
+    );
+  } catch (e) {
+    if (e instanceof AbortError) {
+      return;
+    }
+    throw e;
+  }
+}
+
+export async function collectIDBDatabases(
+  idbDatabasesStore: IDBDatabasesStore,
+  signal: AbortSignal,
+  now: number,
+  maxAge: number,
+  newDagStore = defaultNewDagStore,
+): Promise<void> {
+  const databases = await idbDatabasesStore.getDatabases();
+
+  const dbs = Object.values(databases) as IndexedDBDatabase[];
+  const canCollectResults = await Promise.all(
+    dbs.map(
+      async db =>
+        [
+          db.name,
+          await canCollectDatabase(db, now, maxAge, newDagStore),
+        ] as const,
+    ),
+  );
+
+  const namesToRemove = canCollectResults
+    .filter(result => result[1])
+    .map(result => result[0]);
+
+  // Try to remove the databases in parallel. Don't let a single reject fail the
+  // other ones. We will check for failures afterwards.
+  const dropStoreResults = await Promise.allSettled(
+    namesToRemove.map(async name => {
+      await dropStore(name);
+      return name;
+    }),
+  );
+
+  const idbRemovedNames: string[] = [];
+  const errors: unknown[] = [];
+  for (const result of dropStoreResults) {
+    if (result.status === 'fulfilled') {
+      idbRemovedNames.push(result.value);
+    } else {
+      errors.push(result.reason);
+    }
+  }
+
+  if (idbRemovedNames.length && !signal.aborted) {
+    // Remove the database name from the meta table.
+    await idbDatabasesStore.deleteDatabases(idbRemovedNames);
+  }
+
+  if (errors.length) {
+    throw errors[0];
+  }
+}
+
+function defaultNewDagStore(name: string): dag.Store {
+  const perKvStore = new kv.IDBStore(name);
+  return new dag.StoreImpl(perKvStore, dag.throwChunkHasher, assertNotTempHash);
+}
+
+async function canCollectDatabase(
+  db: IndexedDBDatabase,
+  now: number,
+  maxAge: number,
+  newDagStore: typeof defaultNewDagStore,
+): Promise<boolean> {
+  if (db.replicacheFormatVersion > REPLICACHE_FORMAT_VERSION) {
+    return false;
+  }
+
+  // 0 is used in testing
+  if (db.lastOpenedTimestampMS !== undefined) {
+    return now - db.lastOpenedTimestampMS >= maxAge;
+  }
+
+  // For legacy databases we do not have a lastOpenedTimestampMS so we check the
+  // time stamps of the clients
+  const perdag = newDagStore(db.name);
```

Comment:
Should we wrap the attempt to read legacy dbs in a try catch to be robust against current code having a conflict with legacy db format?